homeassistant/components/fibaro/__init__.py

```diff
@@ -242,33 +253,44 @@ def _read_devices(self):
                         self.hub_serial, device.id)
                     self._device_map[device.id] = device
                     self.fibaro_devices[device.mapped_type].append(device)
-                else:
-                    _LOGGER.debug("%s (%s, %s) not used",
-                                  device.ha_id, device.type,
-                                  device.baseType)
+                _LOGGER.debug("%s (%s, %s) -> %s. Prop: %s Actions: %s",
+                              device.ha_id, device.type,
+                              device.baseType, device.mapped_type,
+                              str(device.properties), str(device.actions))
             except (KeyError, ValueError):
                 pass
 
 
-def setup(hass, config):
+def setup(hass, base_config):
     """Set up the Fibaro Component."""
-    hass.data[FIBARO_CONTROLLER] = controller = \
-        FibaroController(config[DOMAIN][CONF_USERNAME],
-                         config[DOMAIN][CONF_PASSWORD],
-                         config[DOMAIN][CONF_URL],
-                         config[DOMAIN][CONF_PLUGINS],
-                         config[DOMAIN])
+    config = base_config.get(DOMAIN)
+    gateways = config.get(CONF_GATEWAYS)
```

Comment:
This looks bad. Don't use `dict.get` twice in a row on consecutive nested keys without default. Not knowing about the config validation, it looks like if the first key would be missing, the first lookup would return `None` and we would get an error here for the second key.

Since we know that those keys will exist in the dict, due to config validation, we don't even need to use `dict.get`. Just do `dict[key]`.

Reply:
Thanks, I fixed it